steward/src/cellars/aave_v2_stablecoin.rs

```diff
@@ -1,77 +1,182 @@
+                    return Err(sp_call_error(format!(
+                        "{}: Sweep 'token': argument is an invalid address: {}",
+                        LOG_PREFIX, params.token
+                    )))
+                }
             };
-            AaveV2StablecoinCellarCalls::ClaimAndUnstakeWithAmount(call).encode()
-        }
-        ClaimAndUnstake(_) => {
-            let call = ClaimAndUnstakeCall {};
-            AaveV2StablecoinCellarCalls::ClaimAndUnstake(call).encode()
-        }
-        Rebalance(params) => {
-            let call = RebalanceCall {
-                new_lending_token: params
-                    .address
-                    .parse::<H160>()
-                    .expect("failed to parse token address"),
-                min_new_lending_token_amount: params.min_new_lending_token_amount.into(),
+            let to = match params.to.parse::<H160>() {
+                Ok(t) => t,
+                Err(_) => {
+                    return Err(sp_call_error(format!(
+                        "{}: Sweep 'to': argument is an invalid address: {}",
+                        LOG_PREFIX, params.to
+                    )))
+                }
             };
-            AaveV2StablecoinCellarCalls::Rebalance(call).encode()
-        }
-        AccruePlatformFees(_) => {
-            let call = AccruePlatformFeesCall {};
-            AaveV2StablecoinCellarCalls::AccruePlatformFees(call).encode()
+            let call = SweepCall { token, to };
+            Ok(AaveV2StablecoinCellarCalls::Sweep(call).encode())
         }
         TransferFees(_) => {
             let call = TransferFeesCall {};
-            AaveV2StablecoinCellarCalls::TransferFees(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::TransferFees(call).encode())
         }
-        SetInputToken(params) => {
-            let call = SetInputTokenCall {
-                token: params
-                    .address
-                    .parse::<H160>()
-                    .expect("failed to parse token address"),
-                is_approved: params.is_approved,
-            };
-            AaveV2StablecoinCellarCalls::SetInputToken(call).encode()
+    }
+}
+
+fn validate_route(results: Vec<Result<H160, &String>>) -> Result<(), Error> {
+    let mut bad_addresses = String::new();
+    for r in results {
+        if let Err(addr) = r {
+            bad_addresses.push_str(&format!(", {}", addr))
         }
-        RemoveLiquidityRestriction(_) => {
-            let call = RemoveLiquidityRestrictionCall {};
-            AaveV2StablecoinCellarCalls::RemoveLiquidityRestriction(call).encode()
+    }
+
+    if !bad_addresses.is_empty() {
+        let mut err_string = "Rebalance 'route': array contains invalid address(s)".to_string();
+        err_string.push_str(&bad_addresses);
+        return Err(sp_call_error(format!("{}: {}", LOG_PREFIX, err_string)));
+    }
+
+    Ok(())
+}
+
+fn validate_swap_params(swap_params: Vec<SwapParams>) -> Result<(), Error> {
```

Comment:
Can we add some test cases for these two params validation functions?

This is also amplifying the fact that in the future adding more cellars this way requires a decent amount of code. Probably going to take some deep thinking to figure out a more elegant solution.

Reply:
Will do. And yes great point...